core/src/main/java/org/apache/iceberg/MetadataTableUtils.java

```diff
@@ -59,10 +59,10 @@ public static Table createMetadataTableInstance(TableOperations originTableOps,
   }
 
   public static Table createMetadataTableInstance(TableOperations originTableOps,
+                                                  String catalogName,
```

Comment:
This was broken before as the name of the metadata table started with its type, not catalog.